xpti/include/xpti_trace_framework.hpp

```diff
@@ -0,0 +1,367 @@
+        ///                       shared object or DLL represented by the handle h
+        ///
+        /// @return <addr>        If the query is successful, the function returns
+        ///                       a valid address
+        /// @return nullptr       If the query fails, nullptr is returned
+        ///
+        xpti_plugin_function_t findFunction(xpti_plugin_handle_t h, const char *symbol) {
+            xpti_plugin_function_t func = nullptr;
+            if (h && symbol) {
+#if defined(_WIN32) || defined(_WIN64)
+                func = GetProcAddress(h, symbol);
+#else
+                func = dlsym(h, symbol);
+#endif
+            }
+            return func;
+        }
+
+        /// @brief Loads a shared object or DLL and returns a plugin handle
+        /// @details This function is a platform independent abstraction for loading
+        /// a DLL or shared object and returns a valid plugin handle if successful.
+        ///
+        /// @param [in]  h        The plugin handle which is required to make this
+        ///                       query
+        /// @param [in] symbol    The symbol that needs to be searched within the
+        ///                       shared object or DLL represented by the handle h
+        ///
+        /// @return <addr>        If the query is successful, the function returns
+        ///                       a valid address
+        /// @return nullptr       If the query fails, nullptr is returned
+        ///
+        xpti_plugin_handle_t loadLibrary(const wchar_t *path, std::string &error) {
+            xpti_plugin_handle_t handle = 0;
+#if defined(_WIN32) || defined(_WIN64)
+            handle = LoadLibraryW(path);
+            if (!handle) {
+                error = getLastError();
+            }
+#else
+            std::wstring str = path;
+            std::string utf8_file = convWChar2MBChar(str);
+            handle = dlopen(utf8_file.c_str(), RTLD_LAZY);
+            if (!handle) {
+                error = getLastError();
+            }
+#endif
+            return handle;
+        }
+
+        xpti::result_t unloadLibrary(xpti_plugin_handle_t h) {
+            xpti::result_t fr = xpti::result_t::XPTI_RESULT_SUCCESS;
+#if defined(_WIN32) || defined(_WIN64)
+            if (!FreeLibrary(h)) {
+                //  Error occured while unloading the share object
+                return xpti::result_t::XPTI_RESULT_FAIL;
+            }
+#else
+            if (dlclose(h)) {
+                //  Error occured while unloading the share object
+                return xpti::result_t::XPTI_RESULT_FAIL;
+            }
+#endif
+            return fr;
+        }
+
+        bool checkTraceEnv() {
+            bool enabled = false;
+            std::wstring env = getEnvironmentVariable("XPTI_TRACE_ENABLE");
+            if (env.empty()) {
+                enabled = true;
+                return enabled;
+            }
+            if (env == L"true") {
```

Comment:
The is primarily due to Windows localization, directly names in Japanese for example will be in wide string format and since the environment variables describe paths to the dispatcher and subscriber, I convert everything to wstring. Since it is only used for environment variables, the  bloat should be incremental/relatively small.

Reply:
@keryell Have removed the use of wstring